### PR TITLE
0d element

### PR DESCRIPTION
**Pull request by endJunction**
This adds a point element to the mesh lib, suitable shape function, and makes it available for the local assemblers.

The reason behind this element is that the boundary of a line mesh consists of point elements and to implement Neumann boundary conditions, for example, integration over such elements must be performed.

The only work-in-progress item is to update the ctest suite.
- [ ] Update ctests.

### CODE REVIEW

File: NumLib/Fem/Integration/IntegrationPoint.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/**
+ * \copyright
```
Comment (wenqing) [inline NumLib/Fem/Integration/IntegrationPoint.h]: Point wise Neumann BC dose not need integration, the value of the BC can be input directly.

**Comment by wenqing** [comment]
Looks like this 0 element is only for the consistency of interface for BC. Why not to use Node directly? Find the nodes in the line element by the search function, and assign the BC values directly from the input data. Besides, the Neumann type BC for node (potin-wise source term) can be directly  put to the input data and added to the RHS vector without any additional processing.

**Comment by TomFischer** [comment]
> Looks like this 0 element is only for the consistency of interface for BC.

This is a good point. We need consistent interfaces and clear structures. The node should not implement the interface of an element.

**Comment by endJunction** [comment]
Zero sized arrays (occuring at some point of creation of 0x0 matrices) are not allowed by c++. This PR needs a general rework.

**Pull request closed without merging**